### PR TITLE
fixed-main-height-calculation

### PR DESCRIPTION
**Pull request by djidji01**
here is my proposal! #3150 , #3137 and #3123

### CODE REVIEW

**Review by danny007in — COMMENTED** [review]
See the bug in top-nav.html file

Screenshot 2020-11-26 172344

**Comment by djidji01** [comment]
> See the bug in top-nav.html file,
> its still remains same over there

this is due to the fact that is missplaced title for `.control-sidebar` outside `.control-sidebar-content` . moving it inside solve the issue.

image

**Comment by danny007in** [comment]
> 
> 
> > See the bug in top-nav.html file,
> > its still remains same over there
> 
> this is due to the fact that is missplaced title for `.control-sidebar` outside `.control-sidebar-content` . moving it inside solve the issue.
> 
> image

OK ill trying to fix

File: build/js/Layout.js [hunk 1]
```diff
@@ -83,19 +82,29 @@ class Layout {
...
+          $contentSelector.css(this._config.panelAutoHeightMode, (max + offset) - heights.footer)
         }
-      } else if (max === heights.window) {
-        $contentSelector.css(this._config.panelAutoHeightMode, (max + offset) - heights.header - heights.footer)
       } else {
         $contentSelector.css(this._config.panelAutoHeightMode, (max + offset) - heights.header)
       }
 
       if (this._isFooterFixed()) {
-        $contentSelector.css(this._config.panelAutoHeightMode, parseFloat($contentSelector.css(this._config.panelAutoHeightMode)) + heights.footer)
```
Comment (danny007in) [inline build/js/Layout.js:98]: both are same

**Comment by REJack** [comment]
The calculation was fixed with @danny007in's PR #3158, I see need for this PR. 
I've removed the unnecessary control-sidebar demo content.

**Comment by djidji01** [comment]
> The calculation was fixed with @danny007in's PR #3158, I see need for this PR.

It was not fixed at all My Dear!! 

to prove it just try and remove a single link item from sidebar, toggle the control-sidebar and see the same bug. the behavior you see with   PR #3158 is because of the new **_Tabbed IFrame Plugin_** that make the `($('.sidebar').height() - $('.control-sidebar-content').height()) > $('.footer').height()`

**Comment by REJack** [comment]
I see what you mean, I've added a fix for this in master repo please test it 😄.

**Comment by djidji01** [comment]
> I see what you mean, I've added a fix for this in master repo please test it

Yeap fixed, however: 

- It seems @danny007in commit is not applied. footer is still not at the bottom

- Shouldn't the  `heights.controlSidebar + heights.footer >= heights.sidebar && heights.controlSidebar != 0` check replace `max === heights.controlSidebar`  just to not repeat the same thing.

- shouldn't the `$body.hasClass(CLASS_NAME_LAYOUT_TOP_NAV)` be removed because my cause unexpected behavior since the bag was only due to the misplaced` title content`.

regards

**Review by danny007in — COMMENTED** [review]
> 
> 
> > I see what you mean, I've added a fix for this in master repo please test it
> 
> Yeap fixed, however:
> 
>     * It seems @danny007in commit is not applied. footer is still not at the bottom
> 
>     * Shouldn't the  `heights.controlSidebar + heights.footer >= heights.sidebar && heights.controlSidebar != 0` check replace `max === heights.controlSidebar`  just to not repeat the same thing.
> 
>     * shouldn't the `$body.hasClass(CLASS_NAME_LAYOUT_TOP_NAV)` be removed because my cause unexpected behavior since the bag was only due to the misplaced` title content`.
> 
> 
> regards



> 
> 
> > I see what you mean, I've added a fix for this in master repo please test it
> 
> Yeap fixed, however:
> 
>     * It seems @danny007in commit is not applied. footer is still not at the bottom
> 
>     * Shouldn't the  `heights.controlSidebar + heights.footer >= heights.sidebar && heights.controlSidebar != 0` check replace `max === heights.controlSidebar`  just to not repeat the same thing.
> 
>     * shouldn't the `$body.hasClass(CLASS_NAME_LAYOUT_TOP_NAV)` be removed because my cause unexpected behavior since the bag was only due to the misplaced` title content`.
> 
> 
> regards

**Comment by djidji01** [comment]
> I see what you mean, I've added a fix for this in master repo please test it 😄.



> I see what you mean, I've added a fix for this in master repo please test it 😄.

Also the code will cause unexpected view on.  `. content-wrapper ` when  `max===window. height() `

File: build/js/Layout.js [hunk 2]
```diff
@@ -83,19 +82,29 @@ class Layout {
...
+        $contentSelector.css(this._config.panelAutoHeightMode, parseFloat($contentSelector.css('min-height')) + heights.footer)
```
[the remark this answers is not in the document]
Comment (djidji01) [reply build/js/Layout.js:107]: Sure.

**Comment by REJack** [comment]
@djidji01 please test master branch again (you need to compile the js/css files), the latest PR from @danny007in is merged and now I found no bug.

**Comment by djidji01** [comment]
> 
> 
> @djidji01 please test master branch again (you need to compile the js/css files), the latest PR from @danny007in is merged and now I found no bug.

@REJack  As I said just look what happen when window is the max height!!!

image

**Comment by danny007in** [comment]
> > 
> > 
> > @djidji01 please test master branch again (you need to compile the js/css files), the latest PR from @danny007in is merged and now I found no bug.
> 
> @REJack  As I said just look what happen when window is the max height!!!
> 
> image
> 
> 

Actual compilation master branch will be before new release.

**Comment by REJack** [comment]
Does this bug still exist in the master branch?

**Comment by danny007in** [comment]
> 
> 
> Does this bug still exist in the master branch?

No,
Close it

**Comment by djidji01** [comment]
> 
> 
> > @djidji01 please test master branch again (you need to compile the js/css files), the latest PR from @danny007in is merged and now I found no bug.
> 
> @REJack As I said just look what happen when window is the max height!!!
> 
> image

@danny007in @REJack unless you neglected this and it is still a bug.

thanks

**Comment by danny007in** [comment]
> 
> 
> > > @djidji01 please test master branch again (you need to compile the js/css files), the latest PR from @danny007in is merged and now I found no bug.
> > 
> > 
> > @REJack As I said just look what happen when window is the max height!!!
> > image
> 
> @danny007in @REJack unless you neglected this and it is still a bug.
> 
> thanks

@djidji01 Merge master again and Check now 😄

**Comment by danny007in** [comment]
> 
> 
> > > @djidji01 please test master branch again (you need to compile the js/css files), the latest PR from @danny007in is merged and now I found no bug.
> > 
> > 
> > @REJack As I said just look what happen when window is the max height!!!
> > image
> 
> @danny007in @REJack unless you neglected this and it is still a bug.
> 
> thanks

OK i checked this, Yes its not fixed @djidji01 @REJack

**Comment by danny007in** [comment]
@REJack Not solved 

8U1G35n65J

**Pull request closed without merging**